java/src/main/java/com/spotify/featran/java/SerializableFunction.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.featran.java;
+
+import java.io.Serializable;
+
+public interface SerializableFunction<InputT, OutputT> extends Serializable {
```

Comment:
Not possible to do this with java 8 lambdas because of the serializable part?  Would be nice to use lambdas when specifying the function instead of creating an instance of this interface.  

```(InputT p) -> p.outputT```